src/components/MagicCodeInput.tsx

```diff
@@ -286,7 +286,7 @@ function MagicCodeInput(
 
             // If the currently focused index already has a value, it will delete
             // that value but maintain the focus on the same input.
-            if (focusedIndex !== undefined && numbers?.[focusedIndex] !== CONST.MAGIC_CODE_EMPTY_CHAR) {
+            if (focusedIndex !== undefined && numbers?.at(focusedIndex) !== CONST.MAGIC_CODE_EMPTY_CHAR) {
```

Comment:
Please make sure `focusedIndex` won't be negative since it can affect the logic then.